pkg/extension/workspace/example/envoy/util/load.go

```diff
@@ -30,32 +32,41 @@ import (
 func Load(config *model.File, message proto.Message) error {
 	json, err := yaml.YAMLToJSON(config.Content)
 	if err != nil {
-		return errors.Wrapf(err, "failed to convert into JSON Envoy config coming from %q", config.Source)
+		return fmt.Errorf("failed to convert into JSON Envoy config coming from %q: %w", config.Source, err)
 	}
 
-	unmarshaller := jsonpb.Unmarshaler{
-		AllowUnknownFields: true,
+	unmarshaller := protojson.UnmarshalOptions{
+		DiscardUnknown: true,
 		// ignore unknown extension types that might appear in a user-defined Envoy config
-		AnyResolver: newFakeAnyResolver(),
+		Resolver: fakeAnyResolver{},
 	}
-	return unmarshaller.Unmarshal(bytes.NewReader(json), message)
-}
-
-func newFakeAnyResolver() jsonpb.AnyResolver {
-	return fakeAnyResolver{}
+	return unmarshaller.Unmarshal(json, message)
 }
 
 // fakeAnyResolver resolves any type URL into a wrong but known message type.
 //
-// When using fakeAnyResolver, the actual payload of a google.protobug.Any field
+// When using fakeAnyResolver, the actual payload of a google.protobuf.Any field
 // will be lost during unmarshalling.
 // On the bright side, unmarshalling will never fail due to use of unknown types.
 type fakeAnyResolver struct{}
 
-func (r fakeAnyResolver) Resolve(typeURL string) (proto.Message, error) {
+func (r fakeAnyResolver) FindMessageByName(message protoreflect.FullName) (protoreflect.MessageType, error) {
+	return r.FindMessageByURL(string(message))
+}
+
+func (r fakeAnyResolver) FindMessageByURL(url string) (protoreflect.MessageType, error) {
 	// NOTE: as of github.com/golang/protobuf/jsonpb@v1.3.x, the only way to achieve
```

Comment:
I left this comment alone intentionally even if it refers to an older version. I don't know if there's a way around this in the new version, but I also couldn't find one looking in an hour. This approach still works.